package.json

```
@@ -15,7 +15,8 @@
     "lint:js": "eslint --ext '.js,.ts,.tsx' src/ example/src FabricExample/src MacOSExample/src && yarn prettier --check './{src,example,FabricExample,MacOSExample}/**/*.{js,jsx,ts,tsx}'",
     "lint:js-root": "eslint --ext '.js,.ts,.tsx' src/ && yarn prettier --check './src/**/*.{js,jsx,ts,tsx}'",
     "lint:android": "./android/gradlew -p android spotlessCheck -q",
-    "checkIntegrity": "(cd ./FabricExample/android && ./gradlew generateCodegenArtifactsFromSchema -PskipCodegenCopyTask) && (cd ./android && ./gradlew checkIntegrityBetweenArchitectures)"
+    "checkIntegrity": "(cd ./FabricExample/android && ./gradlew generateCodegenArtifactsFromSchema -PskipCodegenCopyTask) && (cd ./android && ./gradlew checkIntegrityBetweenArchitectures)",
+    "checkForCircularDependencies": "yarn madge --extensions js,ts,tsx --circular src"
```

```suggestion
    "circularDependencyCheck": "yarn madge --extensions js,ts,tsx --circular src"
```

I think it is enough, we don't have to overcomplicate that name. Also I think using `-` as separator is more readable than camelCase, but since we have `checkIntegrity` script we can leave that (or we can change both, cc @j-piasecki)

---

Yep, i agree but preferred to follow the convention instead of having both cases in a single filename next to eachother. 

Same for the name, but since you prefer the latter one, I'll change it.

---

Okay, I get what you mean by "same for the name" 😅 Currently we have only one script that starts with `check` so I think it is a bit different compared to `-` and camelCase. Starting with `check` prefix isn't that bad, but having many scripts starting with the same prefix would make `tab` autocomplete a bit frustrating 😅

This is a topic for discussion and I think that following `Reanimated` convention would be better, but let's hear @j-piasecki opinion on that.

---

I think we can merge it as is, but create a follow-up that will take care of this, given that we already have a `ts-check` script (to use dashes instead of camelCase).

---

https://github.com/software-mansion/react-native-gesture-handler/pull/2973
@j-piasecki